vm/src/builtins.rs

```diff
@@ -613,6 +613,24 @@ impl Printer for std::io::StdoutLock<'_> {
     }
 }
 
+pub fn builtin_exit(vm: &VirtualMachine, args: PyFuncArgs) -> PyResult {
+    arg_check!(vm, args, required = [], optional = [(object, None)]);
```

Comment:
Instead of using `arg_check!()`, we have a new system where we can do type checking based on the function declaration's parameters. I'd recommend making the function signature be:

```rust
pub fn builtin_exit(object: OptionalArg<PyObjectRef>, vm: &VirtualMachine) -> PyResult<()>
```

That will handle optional arg checking for you.

Reply:
Thanks !
I am wondering why some functions take a &VirtualMachine as the first argument and others as the last argument.
Is there any rule for ordering the arguments ?

Reply:
The functions that take `&VirtualMachine` as their first argument are all old-style python function definitions. Notice how they're all a `fn(&VirtualMachine, PyFuncArgs)` and use the `arg_check!` macro. The newer style is to specify argument types right in the function signature itself, and then have the `FromArgs` implementation for the type handle the type-checking. If it's a more complex function signature that you need to represent, e.g. with named arguments, you can use the `#[derive(FromArgs)]` procedural macro, a good example of which is the [`PrintOptions` struct](https://github.com/RustPython/RustPython/blob/8359c6e0310f2019b9e0618155d77e2a7b4c9074/vm/src/builtins.rs#L574-L584). The `&VirtualMachine` argument coming before or after doesn't really matter much, it's just that that's how it happened to be implemented for each style of function declaration.